Framework/MDAlgorithms/src/IntegratePeaksMD2.cpp

```diff
@@ -470,9 +470,15 @@ template <typename MDE, size_t nd> void IntegratePeaksMD2::integrate(typename MD
   // Initialize progress reporting
   int nPeaks = peakWS->getNumberPeaks();
   Progress progress(this, 0., 1., nPeaks);
+  bool do_parallel = Kernel::threadSafe(*ws, *peakWS);
+  if (cylinderBool) {
+    do_parallel = do_parallel && Kernel::threadSafe(*wsProfile2D, *wsFit2D, *wsDiff2D);
+  }
+  PARALLEL_FOR_IF(do_parallel)
   for (int i = 0; i < nPeaks; ++i) {
+    PARALLEL_START_INTERRUPT_REGION
     if (this->getCancel())
-      break; // User cancellation
+      throw CancelException();
```

Comment:
What made you change the `break` statement to this `throw` statement?

I'm not sure that L480 and 481 are needed. In other parallel for loops I've seen using OpenMP the cancellation is handled in the PARALLEL_CHECK_INTERRUPT_REGION macro you've got lower down. Since the OpenMP macros have changed since this algorithm originally had a go at being multi-threaded perhaps it was necessary to have the `if (this->getCancel())` back then but don't think it is any more

Reply:
The break statement caused this error
```
12:03:58 /jenkins_workdir/workspace/pull_requests-conda-linux/Framework/MDAlgorithms/src/IntegratePeaksMD2.cpp:477:7: error: break statement used with OpenMP for loop
12:03:58   477 |       break; // User cancellation
12:03:58       |       ^~~~~
12:03:58 ninja: build stopped: subcommand failed.
```
I got the `throw CancelException();` code from `LoadDNSEvent.cpp` which also has this `PARALLEL_CHECK_INTERRUPT_REGION` macro - but I'll remove it and see if I can still cancel it whilst it's running!